src/sentry/tasks/low_priority_symbolication.py

```diff
@@ -0,0 +1,109 @@
+"""
+Tasks that automate the job of moving projects in and out of symbolicator's low priority queue based
+on symbolication metrics stored in Redis.
+
+This has three major tasks, executed in the following general order:
+1. Scan for new suspect projects in Redis that need to be checked for LPQ eligibility. Triggers 2 and 3.
+2. Determine a project's eligibility for the LPQ based on their recorded metrics.
+3. Remove some specified project from the LPQ.
+"""
+
+import logging
+from typing import Iterable
+
+from sentry.tasks.base import instrumented_task
+from sentry.utils import redis
+
+logger = logging.getLogger(__name__)
+
+# redis key for entry storing current list of LPQ members
+LPQ_MEMBERS_KEY = "fill with appropriate value here"
+# key for cluster that stores list of current LPQ members. may be equivalent to CLUSTER_KEY.
+LPQ_CONFIG_CLUSTER_KEY = "fill with appropriate value here"
+
+# key for cluster used to store metrics for LPQ
+CLUSTER_KEY = "fill with appropriate value here"
+
+
+@instrumented_task(name="sentry.tasks.symbolicator.scan_for_suspect_projects", queue="symbolicator.compute_low_priority_queue", ignore_result=True)  # type: ignore
+def scan_for_suspect_projects() -> None:
+    _scan_for_suspect_projects()
+
+
+def _scan_for_suspect_projects() -> None:
+    # todo: this is probably going to be some class wrapper around the cluster to abstract away most
+    # of the nitty-gritty redis work
+    cluster: redis._RedisCluster = redis.get_cluster_from_options(CLUSTER_KEY)
+
+    suspect_projects = set()
+    for item in cluster.scan_iter(
+        # assuming the format is "symbolicate_event_lpq:<project_id>:<posix timestamp or other>"
+        match="symbolicate_event_lpq:*",
+        count=1000,
```

Comment:
it's a loosey goosey attempt to ensure that every scan returns a nonzero number of results as the match gets applied to the result set after pulling them from redis, which means that there's a chance that the match filters out all of the keys. the default count is at 10, which makes this problem more likely to occur with the default values.

in retrospect pretty much all of keys that would be present in this cluster would match this pattern so this shouldn't be a problem, but i've removed the count for a different reason: `scan_iter` should be doing the work for us already, yielding only items that match our search so this count thing isn't too much of a problem.

Reply:
oh right, scan has weird semantics.  I was expecting the `scan_iter` function to handle this for us.. and looking at the code i think it does so, take this example:
```
source = [[0, 1], [], [3, 4, 5]]
def do_iter():
    for batch in source:
        yield from batch
print(list(do_iter()))
```
This results in a single list: `[0, 1, 3, 4, 5]`.   So all the `count` arg does is increase the batch size.  This is probably still a useful thing!  a batch size of 10 is really small, we expect to have about 90 keys per project in this scan, times however many projects we have.  So this could return us about 100 projects at a time.  Keys are about 42 bytes, so a 1000 keys is about 42 Kb of data to transfer in one request.  That still seems reasonable so a batch size of 1000 is probably ok.